bin/zhtautau/runZHTauTauAnalysis.cc

```diff
@@ -2058,29 +2170,39 @@ int main(int argc, char* argv[])
                     mon.fillHisto("eventflow"           , chTagsMain, 8, weight);
                     mon.fillHisto("eventflowNoWeights"  , chTagsMain, 8, 1);
                     mon.fillHisto("eventflowNoLepSF"    , chTagsMain, 8, weightNoLepSF);
+                    mon.fillHisto("eventflow_hMC"       , chTagsMain, 8, decayType, weightNoLepSF);
+                    mon.fillHisto("eventflow_ZMC"       , chTagsMain, 8, ZbosonType, weightNoLepSF);
+
                     mon.fillHisto("nbtags"    , chTags, nbtags,  weight);
                     mon.fillHisto("njets"     , chTags, njets,   weight);
 
                     if(passBJetVetoMain){
                       mon.fillHisto("eventflow"           , chTagsMain, 9, weight);
                       mon.fillHisto("eventflowNoWeights"  , chTagsMain, 9, 1);
                       mon.fillHisto("eventflowNoLepSF"    , chTagsMain, 9, weightNoLepSF);
+                      mon.fillHisto("eventflow_hMC"       , chTagsMain, 9, decayType, weightNoLepSF);
+                      mon.fillHisto("eventflow_ZMC"       , chTagsMain, 9, ZbosonType, weightNoLepSF);
 
                       mon.fillHisto("dPhi_AZ"    , chTagsMain, deltaPhi(higgsCand.phi(), zll.phi()),    weight);
                       mon.fillHisto("dPhi_AMet"  , chTagsMain, deltaPhi(higgsCand.phi(), met.phi()),    weight);
                       mon.fillHisto("dPhi_ZMet"  , chTagsMain, deltaPhi(zll.phi(), met.phi()),    weight);
                       mon.fillHisto("met"      	, chTagsMain, met.pt()         , weight);
 
-                      if(passDPhiCut){
+                      if(true){//if(passDPhiCut){
```

Comment:
thanks @clacaputo, my only comment is about this instruction. While it is certainly correct for the SM ZH final state (by looking at the kinematics), I am not sure it is OK also for the BSM. As the baseline selection starts to diverge I am wondering if not repurposing `zhllvv/runZHllvvAnalysis.cc` and implementing the optimization fro SM analysis there.

Reply:
@clacaputo, I've noticed that this strictly speaking this change is not consistent with 

https://github.com/cp3-lltautau/2l2v_fwk/blob/2aff116be89111364b9ed5178ef8b9ad7d774a49/bin/zhtautau/runZHTauTauAnalysis.cc#L2246

as the `higgsCand_SVFitMass` and `classiSVFit_mass` are used donwnstream in the cut flow, assuming that the cut on  ![codecogseqn](https://user-images.githubusercontent.com/5082376/31838474-44b02564-b5dd-11e7-95db-3b35dae0762e.gif) is fulfilled. My suggestion is then to restore the previous behaviour as @Meriem123 follows-up on the code in the SM analysis.